src/AutoRest.CSharp/Mgmt/Output/Resource.cs

```diff
@@ -23,7 +23,10 @@ public Resource(OperationGroup operationGroup, BuildContext context)
             if (isExtension)
             {
                 var parentOperationGroup = operationGroup.ParentOperationGroup(context);
-                parentValue = parentOperationGroup!.Key.ToSingular(false);
+                if (parentOperationGroup != null)
+                {
+                    parentValue = parentOperationGroup.Key.ToSingular(false);
+                }
```

Comment:
How about the following which just takes one line
```suggestion
                parentValue = parentOperationGroup?.Key.ToSingular(false) ?? string.Empty;
```